spark/src/main/scala/org/apache/spark/sql/delta/commands/merge/MergeIntoMaterializeSource.scala

```diff
@@ -286,6 +285,20 @@ trait MergeIntoMaterializeSource extends DeltaLogging with DeltaSparkPlanUtils {
     )
     rdd.persist(storageLevel)
 
+    // WARNING: if eager == false, the source used during the first Spark Job that uses this may
+    // still be inconsistent with source materialized afterwards.
+    // This is because doCheckpoint that finalizes the lazy checkpoint is called after the Job
+    // that triggered the lazy checkpointing finished.
+    // If blocks were lost during that job, they may still get recomputed and changed compared
```

Comment:
This sounds like an issue in Spark. Maybe it's worth filing a ticket for Spark?

Reply:
Maybe yes.
At first glance doesn't look like an easy one to me.
@bart-samwel I think you should raise it on `#spark-nondeterminism-discuss`, we have the best experts on this there.

Reply:
Maybe yes.
At first glance doesn't look like an easy one to me. We would need some Spark Core RDD experts to weigh in here.